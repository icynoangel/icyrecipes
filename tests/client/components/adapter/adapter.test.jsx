import React from 'react';
import ReactDOM from 'react-dom';
import {mount} from 'enzyme';
import Adapter from './../../../../src/client/js/components/adapter/adapter';
import enzymeToJson from 'enzyme-to-json';

describe('<Adapter />', function() {
  this.getComponent = type => {
    return <Adapter counter={this.counter} component={this.component} />;
  };

  beforeEach(() => {
    this.counter = 1;
    this.component = function(props) {
      return <div>{props.counter}</div>;
    };
  });

  it('Should render wrapped component', () => {
    const wrapper = mount(this.getComponent());
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render wrapped component and update its props', () => {
    const wrapper = mount(this.getComponent());
    wrapper.instance().updateProps({counter: 3});
    wrapper.update();
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });
});
