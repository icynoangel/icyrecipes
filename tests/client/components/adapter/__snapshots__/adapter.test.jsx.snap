// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Adapter /> Should render wrapped component 1`] = `
<Adapter
  component={[Function]}
  counter={1}
>
  <Component
    counter={1}
  >
    <div>
      1
    </div>
  </Component>
</Adapter>
`;

exports[`<Adapter /> Should render wrapped component and update its props 1`] = `
<Adapter
  component={[Function]}
  counter={1}
>
  <Component
    counter={3}
  >
    <div>
      3
    </div>
  </Component>
</Adapter>
`;
